api/filter/network/ext_authz.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+package envoy.api.v2.filter.network;
+
+import "google/protobuf/duration.proto";
+
+import "validate/validate.proto";
+
+// [#not-implemented-hide:]
+// External Authorization filter calls out to an external service over the
+// gRPC Authorization API defined by api/external_auth.proto.
+// A failed check will cause this filter to close the TCP connection.
+message ExtAuthz {
```

Comment:
This message looks like a typical gRPC client config. It doesn't have anything specific to auth. Why don't we make a generic config message.

Reply:
Yes, it would be good to start a generic service configuration and embed it in the service specific configuration even if it is the only message.

Reply:
@mattklein123 @wora I will create a new generic GrpcService at https://github.com/envoyproxy/data-plane-api/tree/master/api/
Filters like this or other(s) can reference it in their protos.

Reply:
If we need timeout, retry, etc. I would recommend pulling these structures directly from rds.proto (the route config protos) if possible. This would make it very easy in the future for service calls to interact directly w/ the router to set complex retry/timeout/etc. policies.